src/ida_lunde_naalsund_ex/ex04/walker.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+__author__ = "Ida Lunde Naalsund"
+__email__ = "idna@nmbu.no"
+
+
+import random as rd
+
+
+class Walker:
+    """
+    Simulates the journey of a walker on his way home in a
+    one dimensional world.
+
+    Attributes
+    ----------
+    x : int
+        The walkers current position
+    steps: int
+        Number of steps needed for walker to achieve current position
+
+    """
+    def __init__(self, x0, h):
+        """
+        Parameters
+        ----------
+        x0: int
+            Starting position
+        h: int
+            Home position
+
+        """
+        self.x = x0
+        self.h = h
+        self.steps = 0
+
+    def move(self):
+        """
+        Makes walker move one step left or right. Whether he walks left or
+        right is determined by whether the random number generated is 0 or 1,
+        accordingly.
+
+        """
+        if rd.randint(0, 1) == 0:
+            self.x -= 1
+        else:
+            self.x += 1
+
+        self.steps += 1
+
+    def is_at_home(self):
+        """
+        Checks if the walker is home.
+
+        Returns:
+        -------
+        bool
+            True if the walker is home, False otherwise.
+
+        """
+        return Walker.get_position(self) == self.h
```

Comment:
this is wrong. It works but for the wrong reason. You should use `self.get_position()`. Ask me about it on friday, its a bit hard to explain why your way is wrong in a short text.